src/app.ts

```diff
@@ -14,17 +14,19 @@ import namespaces from './namespaces';
 
 export type AppState = DefaultState;
 
-export type AppContext = RouterContext<AppState, DatasetContext<{
+export type AppRouterContext = DatasetContext<{
```

Comment:
Is there a better name for this?

Reply:
Could keep using the `Custom` prefix in the name?

Reply:
(from `RouterContext<StateT = any, CustomT = {}>`)

Reply:
Hmm, `CustomContext` doesn't sound great...

Reply:
Looking at it another way, this is extending the (already extended) context with `articles` and possibly other dependencies; assuming it's because they need to be injected into the routes, so this has become a Service Locator / DI Container...

Reply:
Good idea, let's go for `ServiceContext`.